pages/api/interview.ts

```diff
@@ -0,0 +1,69 @@
+import { ChatGPTMessage } from "@/constant/constant";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<any>
+) {
+  try {
+    if (req.method === "POST") {
+      const newMsg: any = await chatGPTHandler(req);
+      return res.status(200).json({
+        data: newMsg.choices[0].message,
+      });
+    }
+  } catch (error) {
+    return res.status(401).json({ error });
+  }
+}
+
+const chatGPTHandler = async (req: NextApiRequest): Promise<Response> => {
+  const body = await JSON.parse(req.body);
+
+  const requestHeaders: Record<string, string> = {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${process.env.OPENAI_API_KEY ?? "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"}`,
+  };
+
+  const { skills, yearsOfExperience, jobRole, jobDescription } = body.userInfo;
+
+  const messages: ChatGPTMessage[] = [
+    {
+      role: "system",
+      content: `You are a friendly technical interviewer with name TechTalQ, 
+      and you will take a short mock interview and 
+      will ask total 2 questions and verify answers from 
+      the user and give the feedback to the user after. 
+      Start by welcoming, asking candidate's name and guiding how the process will go. 
+      you are always friendly, kind, and inspiring and thoughtful responses to the user.
+      All questions should be based on ${skills.join(
+        ","
+      )} with ${yearsOfExperience} years of experience and job role is ${jobRole}
+      ${jobDescription ? ` and job description is "${jobDescription}"` : ""}. 
+      `,
+    },
+    {
+      role: "user",
+      content: `Hi`,
+    },
+  ];
+  messages.push(...body?.messages);
+
+  const payload: any = {
+    model: process.env.OPENAI_API_MODEL,
+    messages: messages,
+    temperature: 0.2,
+  };
+
+  const response = await fetch(
+    `${process.env.OPENAI_API_BASE_URL}/v1/chat/completions`,
```

Comment:
Why we didnt use open ai npm package? I am not saying its the wrong approach but was there a reason or we just picked API call randomly?

Reply:
No, there is not any reason behind this we picked randomly but we can use npm package. 

I think we are using only one API so why do we need the whole package?